HandyHansel/Utilities/ExtensionMethods.cs

```diff
@@ -0,0 +1,194 @@
+                        });
+                    }
+                }
+
+                if (discordEvent is MessageReactionAddEventArgs || discordEvent is MessageReactionRemoveEventArgs)
+                {
+                    DiscordEmoji reactEmoji = discordEvent switch
+                    {
+                        MessageReactionAddEventArgs addReact => addReact.Emoji,
+                        MessageReactionRemoveEventArgs deleteReact => deleteReact.Emoji,
+                        _ => throw new Exception("Somehow, something happened that caused an event that I know to be a reaction add or remove to suddenly stop being that. XD.")
+                    };
+
+
+                    if (reactEmoji.Equals(paginationEmojis.SkipLeft))
+                    {
+                        currentPage = 0;
+                    }
+                    else if (reactEmoji.Equals(paginationEmojis.Left))
+                    {
+                        currentPage = (--currentPage < 0, behaviour) switch
+                        {
+                            (true, PaginationBehaviour.Ignore) => 0,
+                            (true, PaginationBehaviour.WrapAround) => pagesList.Count - 1,
+                            _ => currentPage
+                        };
+                    }
+                    else if (reactEmoji.Equals(paginationEmojis.Right))
+                    {
+                        int count = pagesList.Count;
+                        currentPage = (++currentPage == pagesList.Count, behaviour) switch
+                        {
+                            (true, PaginationBehaviour.Ignore) => pagesList.Count - 1,
+                            (true, PaginationBehaviour.WrapAround) => 0,
+                            _ => currentPage
+                        };
+                    }
+                    else if (reactEmoji.Equals(paginationEmojis.SkipRight))
+                    {
+                        currentPage = pagesList.Count - 1;
+                    }
+
+                    if (reactEmoji.Equals(paginationEmojis.Stop))
+                    {
+                        switch (deletion)
+                        {
+                            case PaginationDeletion.DeleteEmojis:
+                                await msg.DeleteAllReactionsAsync();
+                                break;
+                            case PaginationDeletion.DeleteMessage:
+                                await msg.DeleteAsync();
+                                break;
+                            default:
+                                break;
+                        }
+                        return new CustomResult<T>(cancelled: true);
+                    }
+                    else
+                    {
+                        await msg.ModifyAsync(embed: pagesList[currentPage].Embed);
+                    }
+                }
+            }
+        }
+    }
+}
```

Comment:
Add special extension method to allow for doing pagination and wait for a message.